src/mercury_engine_data_structures/version_validation.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+from hashlib import md5
+from typing import TYPE_CHECKING
+
+from mercury_engine_data_structures.formats.toc import Toc
+from mercury_engine_data_structures.game_check import GameVersion
+
+if TYPE_CHECKING:
+    from mercury_engine_data_structures.file_tree_editor import FileTreeEditor
+
+
+def _get_md5(data: bytes) -> bytes:
+    return md5(data, usedforsecurity=False).digest()
+
+
+def identify_version(editor: FileTreeEditor) -> GameVersion:
+    romfs = editor.romfs
+    toc = romfs.get_file(Toc.system_files_name())
+    toc_hash = _get_md5(toc)
+    for ver in GameVersion:
+        if ver.toc_hash == toc_hash:
+            return ver
+
+    raise ValueError("Not a valid version!")
+
+
+def verify_file_structure(editor: FileTreeEditor) -> GameVersion:
```

Comment:
Can you remove this function and `verify_file_integrity`? I'm not happy with how they work and I'd rather have not block this PR because of these.